src/time/span.cr

```diff
@@ -60,13 +60,19 @@ struct Time::Span
   def self.new(days : Int, hours : Int, minutes : Int, seconds : Int, nanoseconds : Int = 0)
     new(
       seconds: compute_seconds!(days, hours, minutes, seconds),
-      nanoseconds: nanoseconds.to_i64,
+      nanoseconds: nanoseconds,
     )
   end
 
-  def initialize(*, seconds : Int, nanoseconds : Int)
+  def initialize(*, seconds : Int8 | Int16 | Int32 | Int64 | UInt8 | UInt16 | UInt32, nanoseconds : Int = 0)
+    seconds = seconds.to_i64
     # Normalize nanoseconds in the range 0...1_000_000_000
-    seconds += nanoseconds.tdiv(NANOSECONDS_PER_SECOND)
+    # check for possible overflow seconds could become too big
+    sec = nanoseconds.tdiv(NANOSECONDS_PER_SECOND)
```

Comment:
I think that `sec` itself could be `> Int64::MAX` here. That overflow case isn't handled (i think?)

Reply:
It could be, but then `nanoseconds` would need to be really big. `tdiv` returns the result value with the type of `nanoseconds`, so `sec` will have the same possible range as `nanoseconds`. 
But because `sec` will be smaller than `nanoseconds` it fits and doesn't overflow.